src/parser.rs

```diff
@@ -2083,14 +2083,25 @@ impl<'input, 'abbrev, 'unit, Endian> EntriesCursor<'input, 'abbrev, 'unit, Endia
                         unit: self.unit,
                     });
 
-                    Some(Ok(self.cached_current.as_ref().unwrap().clone()))
+                    Some(Ok(self.cached_current.as_ref().unwrap()))
                 } else {
                     Some(Err(Error::UnknownAbbreviation))
                 }
             }
         }
     }
 
+    /// Get the entry that the cursor is currently pointing to.
+    pub fn current<'me>
+        (&'me mut self)
+         -> Option<ParseResult<DebuggingInformationEntry<'input, 'abbrev, 'unit, Endian>>> {
+        match self.current_ref() {
+            Some(Ok(current)) => Some(Ok(current.clone())),
+            Some(Err(e)) => Some(Err(e)),
+            None => None,
```

Comment:
I guess this is more readable than using `map`:

```rust
self.current_ref().map(|res| res.map(|cur| cur.clone()))
```

Are there type errors if we do

```rust
match self.current_ref() {
    Some(Ok(current)) => Some(Ok(current.clone())),
    otherwise => otherwise,
}
```

instead? I guess the type of `otherwise` will be `Option<ParseResult<&DebuggingInformationEntry>>` and not `Option<ParseResult<DebuggingInformationEntry>>` even though we know it doesn't actually contain the variant that differs between the two.